reactions/src/SPT.cpp

```diff
@@ -652,8 +651,29 @@ static double ploopn2_mgdd( const PowerSpectrum& P_L, double vars[], double k, d
     return pow2(r)*2.*P_L(k*r)*(P_L(kargs[0])*p22 + 3.*P_L(k)*p13);
 }
 
+// BILL MOD
+static double ploopn2_mgdd_nu( const PowerSpectrum& P_L, double vars[], double k, double r, double x){
+  double kargs[4],kv[3],xv[3], p22,p13;
+  IOW iow;
+        kv[0] = k;
+        kv[1] = k*r;
+        kv[2] = kv[1];
+        xv[0] = -0.99999999; // don't set to -1 otherwise solver encounters singularity
```

Comment:
I'd write this as -1.0 + tol, and define tol as a tolerance parameter for the solver.

Reply:
Changed all instances in SPT.cpp and added a comment